pkg/apis/serving/metadata_validation.go

```diff
@@ -63,6 +63,20 @@ func validateKnativeAnnotations(annotations map[string]string) (errs *apis.Field
 	return
 }
 
+const noAutoscalingMsg = "annotations in group 'autoscaling.knative.dev' won't work here, they must be put under 'spec.template.metadata.annotations' instead"
+
+// ValidateHasNoAutoscalingAnnotation validates that the respective entity does not have
+// annotations from the autoscaling group. It's to be used to validate Service and
+// Configuration.
+func ValidateHasNoAutoscalingAnnotation(annotations map[string]string) *apis.FieldError {
+	for key := range annotations {
+		if strings.HasPrefix(key, autoscaling.GroupName) {
+			return apis.ErrGeneric(noAutoscalingMsg, apis.CurrentField)
```

Comment:
I wish we had `warnings` -- it's not really an error to set them here, it's just they are useless.